apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -203,4 +216,18 @@ private void sendMessage(JSONObject json) {
     // TODO: check on size of message and truncation etc (this can come later if/when we add more)
     log.debug("Published SNS Message - " + publishResult.getMessageId());
   }
+
+  Map<String, String> getHkaasParams(Map<String, String> parameters) {
+    Pattern pattern = Pattern.compile(HKAAS_REGEX);
```

Comment:
Compiling a regex can be an expensive operation, it probably doesn't make much difference here but it's generally good practice to compile it once and then use it for matching multiple times. So this should be compiled once if the param is set (I guess in the constructor?)

Reply:
+1 plus we need to decide what we do if the regex is invalid, RuntimeException? log and ignore?

Reply:
Moved to constructor. If the variable is not set, then it will not be used. If the regex is invalid, then following the concept of fail-fast, it will throw a `PatternSyntaxException` and Listener will fail to load causing hive-hcatalog-server to not start. I think this is good as then the user will be able to fix the regex without any unintended consequences.